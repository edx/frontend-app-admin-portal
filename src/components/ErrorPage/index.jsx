import React from 'react';
import PropTypes from 'prop-types';
import Helmet from 'react-helmet';

import StatusAlert from '../StatusAlert';
import H1 from '../H1';
import NotFoundPage from '../NotFoundPage';
import ForbiddenPage from '../ForbiddenPage';


function renderErrorComponent(status, message) {
  const errorMessage = message || 'An unknown error has occured.';
  if (status === 404) {
    return <NotFoundPage />;
  } else if (status === 403) {
    return <ForbiddenPage />;
  }
  return (
    <React.Fragment>
      <Helmet>
        <title>Error</title>
      </Helmet>
      <div className="row mt-4">
        <div className="col">
          <H1>Error</H1>
          <StatusAlert
            alertType="danger"
            message={errorMessage}
          />
        </div>
      </div>
    </React.Fragment>
  );
}

const ErrorPage = (props) => {
  const { status, message } = props;
  return (
    <main role="main">
      <div className="container-fluid">
        {renderErrorComponent(status, message)}
      </div>
    </main>
  );
};

ErrorPage.propTypes = {
  status: PropTypes.number,
  message: PropTypes.string,
};

ErrorPage.defaultProps = {
  status: null,
  message: '',
};

export default ErrorPage;
