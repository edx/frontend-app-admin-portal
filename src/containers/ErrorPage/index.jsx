import React from 'react';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';
import Helmet from 'react-helmet';

import StatusAlert from '../../components/StatusAlert';
import H1 from '../../components/H1';

const ErrorPage = (props) => {
  const error = props.location && props.location.state && props.location.state.error;
  const errorMessage = (error && error.message) || 'An unknown error has occured.';

  return (
    <div className="container">
      {error && error.status === 404 ? (
        <Redirect to="/404" />
      ) : (
        <div>
          <Helmet>
            <title>Error</title>
          </Helmet>
          <H1>Error</H1>
          <StatusAlert
            alertType="danger"
            message={errorMessage}
          />
        </div>
      )}
    </div>
  );
};

ErrorPage.propTypes = {
  location: PropTypes.shape({
    state: PropTypes.shape({
      error: PropTypes.shape({
        status: PropTypes.number,
        message: PropTypes.string,
      }),
    }),
  }),
};

ErrorPage.defaultProps = {
  location: {},
};

export default ErrorPage;
