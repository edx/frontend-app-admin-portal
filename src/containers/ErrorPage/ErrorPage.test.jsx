import React from 'react';
import renderer from 'react-test-renderer';
import { Redirect, MemoryRouter } from 'react-router-dom';
import { mount } from 'enzyme';

import ErrorPage from './index';

describe('<ErrorPage />', () => {
  it('renders correctly', () => {
    const location = {
      state: {
        error: {
          status: 500,
          message: 'Something went terribly wrong',
        },
      },
    };

    const tree = renderer
      .create((
        <MemoryRouter>
          <ErrorPage location={location} />
        </MemoryRouter>
      ))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('redirects to /404 for 404 errors', () => {
    const location = {
      state: {
        error: {
          status: 404,
          message: 'Not Found',
        },
      },
    };

    const wrapper = mount((
      <MemoryRouter initialEntries={['/test']}>
        <ErrorPage location={location} />
      </MemoryRouter>
    ));
    const expectedRedirect = <Redirect to="/404" />;
    expect(wrapper.containsMatchingElement(expectedRedirect)).toEqual(true);
  });
});
