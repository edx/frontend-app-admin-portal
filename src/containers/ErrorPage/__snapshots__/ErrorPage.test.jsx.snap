// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorPage /> renders correctly 1`] = `
<div
  className="container"
>
  <div>
    <h1>
      Error
    </h1>
    <div
      className="alert fade alert-danger show"
      hidden={false}
      role="alert"
    >
      <div
        className="alert-dialog"
      >
        <div
          className=""
        >
          <div
            className="message"
          >
            <span>
              Something went terribly wrong
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
